utils/kubernetes/client.go

```diff
@@ -1,70 +1,162 @@
 package kubernetes
 
 import (
-	"os"
-	"path/filepath"
+    "os"
+    "path/filepath"
+
+    "github.com/layer5io/meshkit/models"
+    "gopkg.in/yaml.v2"
+    "github.com/layer5io/meshkit/utils"
+    events "github.com/layer5io/meshkit/utils/events"
+    "k8s.io/client-go/rest"
+    "k8s.io/client-go/tools/clientcmd"
+    clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+    "log"
 
-	"github.com/layer5io/meshkit/utils"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
-	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
 )
 
+// CustomEvent is a custom event type for recording events.
+type CustomEvent struct {
+    EventType string
+    Level     string
+    Message   string
+}
+
 // DetectKubeConfig detects the kubeconfig for the kubernetes cluster and returns it
 func DetectKubeConfig(configfile []byte) (config *rest.Config, err error) {
-	if len(configfile) > 0 {
-		cfgFile, err := processConfig(configfile)
-		if err != nil {
-			return nil, err
-		}
-
-		if config, err = clientcmd.RESTConfigFromKubeConfig(cfgFile); err == nil {
-			return config, err
-		}
-	}
-
-	// If deployed within the cluster
-	if config, err = rest.InClusterConfig(); err == nil {
-		return config, err
-	}
-
-	// Look for kubeconfig from the path mentioned in $KUBECONFIG
-	kubeconfig := os.Getenv("KUBECONFIG")
-	if kubeconfig != "" {
-		if config, err = clientcmd.BuildConfigFromFlags("", kubeconfig); err == nil {
-			return config, err
-		}
-	}
-
-	// Look for kubeconfig at the default path
-	path := filepath.Join(utils.GetHome(), ".kube", "config")
-	if config, err = clientcmd.BuildConfigFromFlags("", path); err == nil {
-		return config, err
-	}
-
-	return
+    if len(configfile) > 0 {
```

Comment:
@yash37158 if you can eradicate the extraneous formatting changes, this will really help reviewers target considering your actual changes. Otherwise, review will take longer.

Reply:
yeah sure e